pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentObserver.java

```diff
@@ -84,6 +84,9 @@ public void onNext(MailboxContent mailboxContent) {
           cancelStream();
           break;
         case FIRST_ERROR:
+          _responseObserver.onNext(MailboxStatus.newBuilder().setMailboxId(mailboxId)
```

Comment:
Why do we need to send back an early termination signal on receiving an error block? The error block would be coming from the sender side itself and so it should be able to do whatever cleanup is required, right?

Reply:
> Why do we need to send back an early termination signal on receiving an error block? The error block would be coming from the sender side itself and so it should be able to do whatever cleanup is required, right?

Yes, it should. But we set the early termination just in case there is a bug on the sending side. Once https://github.com/apache/pinot/pull/16899 it shouldn't be necessary, but TBH it doesn't hurt as well and may prevent leaks if we erroneously change the sending code.